File: src/main/java/com/wanderers/hotelier_webservices/server/service/impl/HotelierServiceImpl.java
package com.wanderers.hotelier_webservices.server.service.impl;

import com.wanderers.hotelier_webservices.server.dao.api.HotelierDao;
import com.wanderers.hotelier_webservices.server.exception.HotelierServiceException;
import com.wanderers.hotelier_webservices.server.service.api.HotelierService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service("hotelier_service")
public class HotelierServiceImpl implements HotelierService {

    private final HotelierDao hotelierDao;

    @Autowired
    HotelierServiceImpl(HotelierDao hotelierDao) {
        this.hotelierDao = hotelierDao;
    }

    @Override
    public Boolean isExistingHotelier(String hotelierId) throws HotelierServiceException {
        try {
            return hotelierDao.isExistingHotelier(hotelierId);
        } catch (Exception e) {
            throw new HotelierServiceException("Failed retrieving hotelier by id", e);
        }
    }
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/service/impl/HotelierServiceImpl.kt
package com.wanderers.hotelier_webservices.server.service.impl

import com.wanderers.hotelier_webservices.server.dao.api.HotelierDao
import com.wanderers.hotelier_webservices.server.exception.HotelierServiceException
import com.wanderers.hotelier_webservices.server.service.api.HotelierService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service("hotelier_service")
class HotelierServiceImpl @Autowired internal constructor(hotelierDao: HotelierDao) : HotelierService {
    private val hotelierDao: HotelierDao

    init {
        this.hotelierDao = hotelierDao
    }

    @kotlin.Throws(HotelierServiceException::class)
    override fun isExistingHotelier(hotelierId: String): Boolean {
        return try {
            hotelierDao.isExistingHotelier(hotelierId)
        } catch (e: Exception) {
            throw HotelierServiceException("Failed retrieving hotelier by id", e)
        }
    }
}